theano/tensor/raw_random.py

```diff
@@ -379,10 +379,14 @@ def _infer_ndim_bcast(ndim, shape, *args):
         ndim = template.ndim
     else:
         v_shape = tensor.as_tensor_variable(shape)
-        if ndim is None:
-            ndim = tensor.get_vector_length(v_shape)
-        bcast = [False] * ndim
 
+    if v_shape.ndim != 1:
+        raise TypeError("shape must be a vector or list of scalar, got '%s'" %
+                        v_shape)
+
+    if ndim is None:
+        ndim = tensor.get_vector_length(v_shape)
+        bcast = [False] * ndim
```

Comment:
This line should be outside of the `if` block, otherwise `bcast` does not get defined.